controllers/storagerequest/storagerequest_controller.go

```diff
@@ -208,7 +208,8 @@ func (r *StorageRequestReconciler) initPhase() error {
 		return fmt.Errorf("no storage consumer owner ref on the storage class request")
 	}
 
-	if err := r.get(r.storageConsumer); err != nil {
+	// unless storage request is marked for deletion we need to verify presence of storageconsumer
+	if err := r.get(r.storageConsumer); (errors.IsNotFound(err) && r.StorageRequest.DeletionTimestamp.IsZero()) || err != nil {
```

Comment:
The condition described by this in is very hard to understand, it conflates between 2 different cases, one where there is a non-not-found error and one where there is no error. 

There is even a subtle bug here where we can return a nil from this func without running the entire reconcile (where we should return an error) in the case that the timestamp is non-zero.

To add to the above, I think that ignoring a storage request that is in deletion is wrong as you might need the reconcile to happen fully in order to finalize the deletion process.  

I would suggest rethinking the approach or at least re-write the condition (maybe split it) in a way that it will be easier to understand and reevaluate

Reply:
> it conflates between 2 different cases
- my intention is only one statement. Get storageconsumer unless storagerequest is set for deletion

> we can return a nil from this func without running the entire reconcile 
- we are not doing it

> I think that ignoring a storage request that is in deletion is wrong
- I'm not doing that

> at least re-write the condition (maybe split it)
- ack, as mentioned in the commit msg, the cond is "return err if (consumer is not found when storagerequest exists) or (err not nil)"

As commented earlier, I think there is no need to get the storageconsumer in storagerequest controller.

Reply:
refreshed.